hazelcast/src/main/java/com/hazelcast/map/impl/proxy/MapProxyImpl.java

```diff
@@ -1044,11 +1034,17 @@ public String toString() {
                 new NodeQueryCacheEndToEndConstructor(request));
     }
 
-    private void handleHazelcastInstanceAwareParams(Object... objects) {
+    /**
+     * Handles injection of managed context into the provided {@code objects}
+     * if any object implements a context-aware interface such as
+     * {@link com.hazelcast.core.HazelcastInstanceAware} or {@link com.hazelcast.spi.NodeAware}.
+     *
+     * @param objects the objects which should have the contexts injected
+     */
+    private void initializeManagedContext(Object... objects) {
```

Comment:
The problem with this method is that you will get an array of objects created for every invocation. I would rather see a copy pasted version of this method with 1 argument and one with 2 arguments since that is the common case.

Reply:
I have changed it just to provide the one-parameter method since I don't think it's such a big problem of invoking the method twice for two user objects.